### PR TITLE
First optimize

### PR DESCRIPTION
**Pull request by agutierrez-eb**

### CODE REVIEW

File: invoicing/settings.py [hunk 1]
```diff
@@ -12,7 +12,7 @@
 """
 
 # Build paths inside the project like this: os.path.join(BASE_DIR, ...)
-import os
+import os, sys
```
Comment (gmarchetta-eb) [inline invoicing/settings.py:15]: Is this import needed?

File: invoicing/settings.py [hunk 2]
```diff
@@ -108,7 +108,10 @@
         'PASSWORD': '',
         'HOST': '127.0.0.1',
         'PORT': 3306,
-    }
+        # 'TEST': {
```
Comment (gmarchetta-eb) [inline invoicing/settings.py:111]: Any reason to keep this commented out? If you are not using it anymore, it's better to remove it

File: invoicing_app/management/commands/generate_tax_receipts_new.py [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+
+import pytz
+
+from datetime import datetime as dt
+from dateutil.relativedelta import relativedelta
+
+from invoicing import settings
+
+from invoicing_app.models import PaymentOptions, Event, Order
+from memory_profiler import profile
```
Comment (gmarchetta-eb) [inline invoicing_app/management/commands/generate_tax_receipts_new.py:15]: Should you remove the memory profiler from master? Are you planning to keep it on the codebase to show results after the demo? If so, you should add a backlog item to remove it once it's no longer needed

File: invoicing_app/management/commands/generate_tax_receipts_old.py [hunk 4]
```diff
@@ -13,6 +13,8 @@
 
 from invoicing_app.models import PaymentOptions, Event, Order
 
+from memory_profiler import profile
```
Comment (gmarchetta-eb) [inline invoicing_app/management/commands/generate_tax_receipts_old.py:16]: ditto about keeping profiler in master

File: invoicing_app/management/commands/generate_tax_receipts_old.py [hunk 5]
```diff
@@ -84,13 +86,21 @@ class Command(BaseCommand):
...
+        )
     )
 
     def __init__(self, *args, **kwargs):
         self.logger = logging.getLogger('financial_transactions')
         self.event_id = None
         self.user_id = None
         self.sentry = logging.getLogger('sentry')
+        self.test_set = {}
+        # self.test_set.clear()
```
Comment (gmarchetta-eb) [inline invoicing_app/management/commands/generate_tax_receipts_old.py:103]: Remove commented code

File: invoicing_app/management/commands/generate_tax_receipts_old.py [hunk 6]
```diff
@@ -353,4 +364,6 @@ def generate_tax_receipts(
             self.call_service(orders_kwargs)
 
     def call_service(self, orders_kwargs):
-        print(orders_kwargs)
+        if self.test:
+            # caca = {orders_kawargs['tax_receipt']['event_id']: orders_kwargs}
```
Comment (gmarchetta-eb) [inline invoicing_app/management/commands/generate_tax_receipts_old.py:368]: Remove commented code

File: invoicing_app/management/commands/generate_tax_receipts_old.py [hunk 7]
```diff
@@ -185,14 +198,16 @@ def localice_date(self, country_code, date):
...
     def generate_tax_receipt_event(self, payment_option, event):
-        localize_start_date = str(self.localice_date(
-            payment_option.epp_country,
-            self.period_start
-        ))
-        localize_end_date = str(self.localice_date(
-            payment_option.epp_country,
-            self.period_end
-        ))
+        # localize_start_date = str(self.localice_date(
```
Comment (gmarchetta-eb) [inline invoicing_app/management/commands/generate_tax_receipts_old.py:201]: Remove commented code

File: requirements.txt [hunk 8]
```diff
@@ -21,4 +21,5 @@ travis==0.0.3
 urllib3==1.25.9
 mock==3.0.5
 pytz==2019.3
-ipdb==0.13.2
+memory-profiler==0.57.0
```
Comment (gmarchetta-eb) [inline requirements.txt:24]: If you remove memory profiler references, you should remove this dependency

**Comment by gmarchetta-eb** [comment]
A general comment. 

1. Your PRs should be smaller and focused to a measurable change. If something breaks as a part of this PR there are a lot of things going on and you will have a hard time finding the root cause.
2. Your PR title and (missing) change description should be more descriptive. It will take you more time to create your PR but it will give a reviewer context about what you are trying to achieve, so they have an easier time reviewing your code. Most of the time, reviewers won't have full context on what you are trying to do, so any additional info is always welcome.

File: invoicing_app/management/commands/generate_tax_receipts_new.py [hunk 9]
```diff
@@ -0,0 +1,321 @@
...
+    def __init__(self, *args, **kwargs):
+        self.logger = logging.getLogger('financial_transactions')
+        self.event_id = None
+        self.user_id = None
+        self.sentry = logging.getLogger('sentry')
+        self.test_set = {}
+
+        super(Command, self).__init__(*args, **kwargs)
+
+    def handle(self, **options):
```
Comment (gmarchetta-eb) [inline invoicing_app/management/commands/generate_tax_receipts_new.py:106]: You may want to consider breaking this function into smaller chunks that make specific tasks. As far as I can see, it's currently:

- Resolving today_date
- Resolving declarable_tax_receipt_countries
- Execute query
- Process query results

If you keep your functions at a minimum, ideally doing one thing, code is easier to maintain and test. You could add specific tests for each one of these and make sure that they work well in isolation with different inputs (positive and negative tests), and then write one integrated test for the handle function itself.

File: invoicing_app/models.py [hunk 10]
```diff
@@ -36,6 +36,10 @@
 
 class Order(models.Model):
 
+    class Meta:
+        managed = True
```
Comment (juanvallejo-eb) [inline invoicing_app/models.py:40]: The managed attribute defaults as True, so it's not necessary to define it explicitly.

File: invoicing_app/tests/factories/order.py [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+            'pp_date',
+            'changed',
+            'event',
+            'mg_fee',
+            'gross',
+            'eb_tax'
+        )
+
+    status = 100
+    pp_date = str(dt(2020, 3, 8, 0, 0))
```
Comment (juanvallejo-eb) [inline invoicing_app/tests/factories/order.py:22]: Maybe you could use something like `factory.fuzzy.FuzzyDateTime` if you want to generate random Datetimes, also I think using str() is not necessary.

**Pull request merged**